### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by nunpa**
This pull request includes some improvements and code clean-up, in addition to the issues #1 #2 #3  and #7.

Explanations are included inside the code.

### CODE REVIEW

File: iblstudiosbadges/BadgeOne_client.py [hunk 1]
```diff
@@ -299,23 +279,66 @@ def get_award_result_formatted(self, resultdata,congratulations):
...
+        Some servers and some old languages will return "\/" instead "/".
 
+        Arguments:
+            data (str): String to be treated
+            
+        Returns:
+            str: String with reversed slashes
+        """
+
+        return data.replace('\/','/')
```
Comment (nedbat) [inline iblstudiosbadges/BadgeOne_client.py]: This function does nothing, so it isn't needed.  The python string `'\/'` is exactly the same as `'/'`.

File: iblstudiosbadges/badges_client.py [hunk 2]
```diff
@@ -129,8 +130,8 @@ def create_obj_badge(self, jsondata,jsonparams):
         """
 
         import json
-        jsonData = json.loads(jsondata, object_hook=self._decode_dict)
-        jsonParams = json.loads(jsonparams, object_hook=self._decode_dict)
+        jsonData = json.loads(jsondata, object_hook=self._auto_encode_dict)
+        jsonParams = json.loads(jsonparams, object_hook=self._auto_encode_dict)
```
Comment (nedbat) [inline iblstudiosbadges/badges_client.py]: I don't understand the purpose of `_auto_encode_dict`.  After it encodes to utf8, most of the accesses of the JSON data decode it from utf8.

**Comment by nedbat** [comment]
Issues #2 and #4 don't seem to be resolved by these changes, but they are fine to merge.

File: iblstudiosbadges/BadgeOne_client.py (hunk 1 above)
Comment (nunpa) [inline iblstudiosbadges/BadgeOne_client.py]: url_test = "http:\/\/www.test.com"
print "Returns => %s" % ( url_test )

url_test = url_test.replace('\/','/')
print "Returns => %s" % ( url_test )

Comment (nedbat) [inline iblstudiosbadges/BadgeOne_client.py]: OK, my mistake :)

**Pull request merged**